tests/python_package_test/test_basic.py

```diff
@@ -217,6 +217,43 @@ def test_sequence_get_data(num_seq):
     np.testing.assert_array_equal(subset_data.get_data(), X[sorted(used_indices)])
 
 
+def test_retrain_list_of_sequence():
+    X, y = load_breast_cancer(return_X_y=True)
+    seqs = _create_sequence_from_ndarray(X, 2, 100)
+
+    seq_ds = lgb.Dataset(seqs, label=y, free_raw_data=False)
+
+    assert sum([len(s) for s in seq_ds.get_data()]) == X.shape[0]
+    assert len(seq_ds.get_feature_name()) == X.shape[1]
+    assert seq_ds.get_data() == seqs
```

Comment:
These checks should be moved after training, to avoid this test failure: https://github.com/microsoft/LightGBM/actions/runs/9935170010/job/27451324230?pr=6414

```text
FAILED tests/python_package_test/test_basic.py::test_retrain_list_of_sequence - Exception: Cannot get data before construct Dataset
```

Please run the tests yourself before pushing another commit.

```shell
sh build-python.sh bdist_wheel install
pytest tests/python_package_tests/test_basic.py::test_retrain_list_of_sequence
```

Reply:
Oh, I only tested the code on jupyter notebook for this case.
Sorry I don't have env for `sh`  , so can not build whell to run the pytest.

Reply:
Ok, I will push testing changes for you.